From b4be67b48fb11989f1824dd0b1c8c399430162b0 Mon Sep 17 00:00:00 2001
From: "Sergey V. Udaltsov" <svu@gnome.org>
Date: Sun, 3 Feb 2013 21:31:07 +0000
Subject: [PATCH] fr --> fra (ISO 639)

---
 rules/base.extras.xml.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rules/base.extras.xml.in b/rules/base.extras.xml.in
index cec043f..4647bec 100644
--- a/rules/base.extras.xml.in
+++ b/rules/base.extras.xml.in
@@ -60,7 +60,7 @@
         <name>ca</name>
         <_shortDescription>fr</_shortDescription>
         <_description>French (Canada)</_description>
-        <languageList><iso639Id>fr</iso639Id></languageList>
+        <languageList><iso639Id>fra</iso639Id></languageList>
       </configItem>
       <variantList>
         <variant>
-- 
1.8.1.4

