From 3f7d33a91dbb8f3ba27d824761364b24e8dac70b Mon Sep 17 00:00:00 2001
From: "Sergey V. Udaltsov" <svu@gnome.org>
Date: Fri, 1 Feb 2013 00:40:51 +0000
Subject: [PATCH] English Mali layouts have English language assigned to them

---
 rules/base.xml.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rules/base.xml.in b/rules/base.xml.in
index 7239c09..19380fa 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -5377,6 +5377,9 @@
             <!-- Keyboard indicator for English layouts -->
             <_shortDescription>en</_shortDescription>
             <_description>English (Mali, US Macintosh)</_description>
+            <languageList>
+              <iso639Id>eng</iso639Id>
+            </languageList>
           </configItem>
         </variant>
         <variant>
@@ -5385,6 +5388,9 @@
             <!-- Keyboard indicator for English layouts -->
             <_shortDescription>en</_shortDescription>
             <_description>English (Mali, US international)</_description>
+            <languageList>
+              <iso639Id>eng</iso639Id>
+            </languageList>
           </configItem>
         </variant>
       </variantList>
-- 
1.8.1.4

