From 7e789cffc6999a30b5d7ded03496b1d3dad1dc86 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 15 Dec 2011 15:53:55 +1000
Subject: [PATCH] rules: Serbian keyboards are cyrillic by default

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 rules/base.xml.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/rules/base.xml.in b/rules/base.xml.in
index 4ee879b..a90e994 100644
--- a/rules/base.xml.in
+++ b/rules/base.xml.in
@@ -4075,7 +4075,7 @@
       <configItem>
         <name>rs</name>
         <_shortDescription>sr</_shortDescription>
-        <_description>Serbian</_description>
+        <_description>Serbian (Cyrillic)</_description>
         <languageList>
           <iso639Id>srp</iso639Id>
         </languageList>
@@ -4084,7 +4084,7 @@
         <variant>
           <configItem>
             <name>yz</name>
-            <_description>Serbian (Z and ZHE swapped)</_description>
+            <_description>Serbian (Cyrillic, Z and ZHE swapped)</_description>
           </configItem>
         </variant>
         <variant>
@@ -4114,7 +4114,7 @@
         <variant>
           <configItem>
             <name>alternatequotes</name>
-            <_description>Serbian (with guillemets)</_description>
+            <_description>Serbian (Cyrillic with guillemets)</_description>
           </configItem>
         </variant>
         <variant>
-- 
1.7.7.1

