diff -up xkeyboard-config-1.2/symbols/kz.kzgroup xkeyboard-config-1.2/symbols/kz
--- xkeyboard-config-1.2/symbols/kz.kzgroup	2008-01-24 18:21:06.000000000 -0500
+++ xkeyboard-config-1.2/symbols/kz	2008-02-07 12:23:56.000000000 -0500
@@ -213,6 +213,7 @@ xkb_symbols "olpc" {
 
    key <TLDE> { [ parenleft,               parenright ] };
    key <BKSL> { [ Cyrillic_io,             Cyrillic_IO ] };
-
+   key <I219> { [ ISO_Next_Group,          ISO_Prev_Group ] };
+ 
    include "group(olpc)"
 };
