diff -u -p -r1.4 ng
--- symbols/ng	28 Oct 2007 08:53:29 -0000	1.4
+++ symbols/ng	20 Nov 2007 06:20:24 -0000
@@ -111,7 +111,7 @@ xkb_symbols "olpc" {
    key <AB09>  { [    period, greater, guillemotright,  guillemotright ] };
    key <AB10>  { [     slash,   question, questiondown,     questiondown ] };
 
-   key <I219>  { [ ISO_Next_Group, ISO_Prev_Group ] };
+   key <I219>  { [  multiply,   division, ISO_Next_Group, ISO_Prev_Group ] };
 
    include "level3(ralt_switch)"
 };
