Index: symbols/af
===================================================================
RCS file: /cvs/xlibs/xkbdesc/xkeyboard-config/symbols/af,v
retrieving revision 1.2
diff -u -p -r1.2 af
--- symbols/af	11 Oct 2006 19:32:35 -0000	1.2
+++ symbols/af	20 Nov 2007 04:49:21 -0000
@@ -211,3 +211,188 @@ xkb_symbols "uz" {
    include "level3(ralt_switch)"
 };
 
+partial alphanumeric_keys 
+xkb_symbols "olpc-ps" {
+
+   name[Group1]= "Afghanistan - Pashto";
+
+   key <TLDE> { [ 0x100200D, 0x1000654, grave ] }; // zero width joiner, Arabic hamza above
+   key <AE01> { [ 0x10006F1, exclam, tilde ] }; // Arabic one
+   key <AE02> { [ 0x10006F2, 0x100066C. at ] }; // Arabic two, Arabic thousands separator
+   key <AE03> { [ 0x10006F3, 0x100066B, numbersign ] }; // Arabic three, Arabic decimal separator
+   key <AE04> { [ 0x10006F4, 0x100060B, dollar ] }; // Arabic four, Afghani sign
+   key <AE05> { [ 0x10006F5, 0x100066A, percent ] }; // Arabic five, Arabic percent sign
+   key <AE06> { [ 0x10006F6, multiply, circumflex ] }; // Arabic six
+   key <AE07> { [ 0x10006F7, guillemontright, ampersand ] }; // Arabic seven
+   key <AE08> { [ 0x10006F8, guillemontleft, 0x100066D ] }; // Arabic eight, Arabic five-pointed star
+   key <AE09> { [ 0x10006F9, parenright, bullet ] }; // Arabic nine
+   key <AE10> { [ 0x10006F0, parenleft, degree ] }; // Arabic zero
+   key <AE11> { [ minus, 0x1000640, underscore ] }; // Arabic_tatweel
+   key <AE12> { [ plus, equal, divide ] };
+
+   key <AD01> { [ 0x1000636, 0x1000652, eurosign ] }; // Arabic dad, Arabic sukun
+   key <AD02> { [ 0x1000635, 0x100064C, 0x1000671 ] }; // Arabic sad, Arabic dammatan, Arabic alef walsa
+   key <AD03> { [ 0x100062B, 0x100064D, 0x1000649 ] }; // Arabic theh, Arabic kasratan, Arabic alef maksura initial form
+   key <AD04> { [ 0x1000642, 0x100064B, 0x100200E ] }; // Arabic qaf, Arabic fathatan, left-to-right mark
+   key <AD05> { [ 0x1000641, 0x100064F, 0x100200F ] }; // Arabic feh, Arabic damma, right-to-left mark
+   key <AD06> { [ 0x100063A, 0x1000650, 0x100e653 ] }; // Arabic ghain, Arabic kasra, Arabic alef with madda above
+   key <AD07> { [ 0x1000639, 0x100064E, 0x100e659 ] }; // Arabic ain, Arabic fatha, Arabic zwarakay
+   key <AD08> { [ 0x1000647, 0x1000651, 0x1000670 ] }; // Arabic heh, Arabic shadda, Arabic superscript alef
+   key <AD09> { [ 0x100062E, 0x1000681, apostrophe ] }; // Arabic khah, Arabic hah with hamza above, 
+   key <AD10> { [ 0x100062D, 0x1000685, quotedbl ] }; // Arabic hah, Arabic hah with three dots above
+   key <AD11> { [ 0x100062C, bracketright, braceleft ] }; // Arabic jeem
+   key <AD12> { [ 0x1000686, bracketleft, braceright ] }; // Arabic tcheh
+
+   key <AC01> { [ 0x1000634, 0x100069A ] }; // Arabic sheen, Arabic seen with dot below and dot above
+   key <AC02> { [ 0x1000633, 0x10006CD ] }; // Arabic seen, Arabic yeh with tail
+   key <AC03> { [ 0x10006CC, 0x100064A, 0x10006D2 ] }; // Farsi yeh, Arabic yeh, Arabic yeh barree
+   key <AC04> { [ 0x1000628, 0x100067E, 0x10006BA ] }; // Arabic beh, Arabic peh, Arabic noon ghunna
+   key <AC05> { [ 0x1000644, 0x1000623, 0x10006B7 ] }; // Arabic lam, Arabic hamza on alef, Arabic alef with hamza above
+   key <AC06> { [ 0x1000627, 0x1000622, 0x1000625 ] }; // Arabic alef, Arabic madda on alef, Arabic alef with hamza below
+   key <AC07> { [ 0x100062A, 0x100067C, 0x1000679 ] }; // Arabic teh, Arabic teh with ring, Arabic tteh
+   key <AC08> { [ 0x1000646, 0x10006BC, greater ] }; // Arabic noon, Arabic noon with ring
+   key <AC09> { [ 0x1000645, 0x1000629, less ] }; // Arabic meem, Arabic teh marbuta
+   key <AC10> { [ 0x10006A9, colon, 0x1000643 ] }; // Arabic keheh, Arabic kaf
+   key <AC11> { [ 0x10006AB, 0x100061B, 0x10006AF ] }; // Arabic kaf with ring, Arabic semicolon, Arabic gaf
+
+   key <BKSL> { [ backslash, asterisk, bar ] };
+   key <SPCE> { [ space, 0x100200C, 0x10000A0 ] }; // zero width non-joiner, no break space
+
+   key <AB01> { [ 0x1000638, 0x1000626, question] }; // Arabic zah, Arabic yeh with hamza above
+   key <AB02> { [ 0x10006D0, 0x1000637, semicolon ] }; // Arabic tah, Arabic E
+   key <AB03> { [ 0x1000632, 0x1000698 ] }; // Arabic zain, Arabic jeh
+   key <AB04> { [ 0x1000631, 0x1000621 ] }; // Arabic_ra (reh?), Arabic hamza
+   key <AB05> { [ 0x1000630, 0x100200C ] }; // Arabic_thal, zero width non-joiner
+   key <AB06> { [ 0x100062F, 0x1000689, 0x1000688 ] }; // Arabic_dal, Arabic dal with ring, Arabic ddal
+   key <AB07> { [ 0x1000693, 0x1000624, 0x1000691 ] }; // Arabic reh with ring, Arabic waw with hamza above, Arabic rreh
+   key <AB08> { [ 0x1000648, period, comma ] }; // Arabic_waw, Arabic comma
+   key <AB09> { [ 0x1000696, 0x100002E, 0x10006C7 ] }; // Arabic reh with dot below and dot above, full stop, Arabic letter U
+   key <AB10> { [ slash, 0x100061F, 0x10006C9 ] }; // Arabic question mark, Arabic kirghiz yu
+
+   //language key
+   key <I219> {  [ISO_Next_Group  ,  ISO_Prev_Group      ]     };
+};
+
+partial alphanumeric_keys 
+xkb_symbols "olpc-da" {
+
+   name[Group1]= "Afghanistan - Dari";
+
+   key <TLDE> { [ 0x100200D, divide, tilde ] }; // zero width joiner
+   key <AE01> { [ 0x10006F1, exclam, grave ] }; // Arabic one
+   key <AE02> { [ 0x10006F2, 0x100066C. at ] }; // Arabic two, Arabic thousands separator
+   key <AE03> { [ 0x10006F3, 0x100066B, numbersign ] }; // Arabic three, Arabic decimal separator
+   key <AE04> { [ 0x10006F4, 0x100060B, dollar ] }; // Arabic four, Afghani sign
+   key <AE05> { [ 0x10006F5, 0x100066A, percent ] }; // Arabic five, Arabic percent sign
+   key <AE06> { [ 0x10006F6, multiply, circumflex ] }; // Arabic six
+   key <AE07> { [ 0x10006F7, 0x100060C, ampersand ] }; // Arabic seven, Arabic comma
+   key <AE08> { [ 0x10006F8, asterisk, bullet ] }; // Arabic eight, 
+   key <AE09> { [ 0x10006F9, parenright, 0x100200E ] }; // Arabic nine, left-to-right mark
+   key <AE10> { [ 0x10006F0, parenleft, 0x100200F ] }; // Arabic zero, right-to-left mark
+   key <AE11> { [ minus, 0x1000640, underscore ] }; // Arabic_tatweel
+   key <AE12> { [ plus, equal ] };
+
+   key <AD01> { [ 0x1000636, 0x1000652, degree ] }; // Arabic dad, Arabic sukun
+   key <AD02> { [ 0x1000635, 0x100064C ] }; // Arabic sad, Arabic dammatan
+   key <AD03> { [ 0x100062B, 0x100064D, eurosign ] }; // Arabic theh, Arabic kasratan
+   key <AD04> { [ 0x1000642, 0x100064B, 0x100FD3E ] }; // Arabic qaf, Arabic fathatan, ornate left paren
+   key <AD05> { [ 0x1000641, 0x100064F, 0x100FD3F ] }; // Arabic feh, Arabic damma, ornate right paren
+   key <AD06> { [ 0x100063A, 0x1000650, 0x1000656 ] }; // Arabic ghain, Arabic kasra, Arabic subscript alef
+   key <AD07> { [ 0x1000639, 0x100064E, 0x100e659 ] }; // Arabic ain, Arabic fatha, Arabic zwarakay
+   key <AD08> { [ 0x1000647, 0x1000651, 0x1000655 ] }; // Arabic heh, Arabic shadda, Arabic hamza below
+   key <AD09> { [ 0x100062E, bracketright, apostrophe ] }; // Arabic khah
+   key <AD10> { [ 0x100062D, bracketleft, quotedbl ] }; // Arabic hah
+   key <AD11> { [ 0x100062C, braceright, 0x1000681 ] }; // Arabic jeem, Arabic hah with hamza above
+   key <AD12> { [ 0x1000686, braceleft, 0x1000685 ] }; // Arabic tcheh, Arabic hah with three dots above
+
+   key <AC01> { [ 0x1000634, 0x1000624, 0x100069A ] }; // Arabic sheen, Arabic waw with hamza above, Arabic seen with dot below and dot above
+   key <AC02> { [ 0x1000633, 0x1000626, 0x10006CD ] }; // Arabic seen, Arabic yeh with hamza above, Arabic yeh with tail
+   key <AC03> { [ 0x10006CC, 0x100064A, 0x1000649 ] }; // Farsi yeh, Arabic yeh, Arabic alef maksura
+   key <AC04> { [ 0x1000628, 0x1000625, 0x10006D0 ] }; // Arabic beh, Arabic alef with hamza below, Arabic e
+   key <AC05> { [ 0x1000644, 0x1000623, 0x10006B7 ] }; // Arabic lam, Arabic hamza on alef, Arabic alef with hamza above
+   key <AC06> { [ 0x1000627, 0x1000622, 0x1000671 ] }; // Arabic alef, Arabic madda on alef, Arabic alef wasla
+   key <AC07> { [ 0x100062A, 0x1000629, 0x100067C ] }; // Arabic teh, Arabic teh marbuta, Arabic tteh
+   key <AC08> { [ 0x1000646, guillemontright, 0x10006BC ] }; // Arabic noon, Arabic noon with ring
+   key <AC09> { [ 0x1000645, guillemontleft, 0x10006BA ] }; // Arabic meem, Arabic noon ghunna
+   key <AC10> { [ 0x10006A9, colon, semicolon ] }; // Arabic keheh, 
+   key <AC11> { [ 0x10006AF, 0x100061B, 0x10006AB ] }; // Arabic gaf, Arabic semicolon, Arabic kaf with ring
+
+   key <BKSL> { [ backslash, bar, question ] };
+   key <SPCE> { [ space, 0x100200C, 0x10000A0 ] }; // zero width non-joiner, no break space
+
+   key <AB01> { [ 0x1000638, 0x1000643, 0x10006D2 ] }; // Arabic zah, Arabic kaf, Arabic yeh barree
+   key <AB02> { [ 0x1000637, 0x1000653, 0x1000691 ] }; // Arabic tah, Arabic maddah above, Arabic rreh
+   key <AB03> { [ 0x1000632, 0x1000698, 0x1000696 ] }; // Arabic zain, Arabic jeh,  Arabic reh with dot below and dot above
+   key <AB04> { [ 0x1000631, 0x1000670, 0x1000693 ] }; // Arabic_ra (reh?), Arabic superscript alef, Arabic reh with ring
+   key <AB05> { [ 0x1000630, 0x100200C, 0x1000688 ] }; // Arabic_thal, zero width non-joiner, Arabic ddal
+   key <AB06> { [ 0x100062F, 0x1000654, 0x1000689 ] }; // Arabic dal, Arabic hamza above, Arabic dal with ring
+   key <AB07> { [ 0x100067E, 0x1000621, 0x1000679 ] }; // Arabic peh, Arabic hamza, Arabic tteh
+   key <AB08> { [ 0x1000648, greater, comma ] }; // Arabic_waw
+   key <AB09> { [ period, less, 0x10006C7 ] }; // Arabic u
+   key <AB10> { [ slash, 0x100061F, 0x10006C9 ] }; // Arabic question mark, Arabic kirghiz yu
+
+   //language key
+   key <I219> { [ ISO_Next_Group,  ISO_Prev_Group ] };
+};
+
+partial alphanumeric_keys 
+xkb_symbols "olpc-uz" {
+
+   name[Group1]= "Afghanistan - Uzbek";
+
+   key <TLDE> { [ 0x100200D, divide, tilde ] }; // zero width joiner
+   key <AE01> { [ 0x10006F1, exclam, grave ] }; // Arabic one
+   key <AE02> { [ 0x10006F2, 0x100066C. at ] }; // Arabic two, Arabic thousands separator
+   key <AE03> { [ 0x10006F3, 0x100066B, numbersign ] }; // Arabic three, Arabic decimal separator
+   key <AE04> { [ 0x10006F4, 0x100060B, dollar ] }; // Arabic four, Afghani sign
+   key <AE05> { [ 0x10006F5, 0x100066A, percent ] }; // Arabic five, Arabic percent sign
+   key <AE06> { [ 0x10006F6, multiply, circumflex ] }; // Arabic six
+   key <AE07> { [ 0x10006F7, 0x100060C, ampersand ] }; // Arabic seven, Arabic comma
+   key <AE08> { [ 0x10006F8, asterisk, bullet ] }; // Arabic eight, 
+   key <AE09> { [ 0x10006F9, parenright, 0x100200E ] }; // Arabic nine, left-to-right mark
+   key <AE10> { [ 0x10006F0, parenleft, 0x100200F ] }; // Arabic zero, right-to-left mark
+   key <AE11> { [ minus, 0x1000640, underscore ] }; // Arabic_tatweel
+   key <AE12> { [ plus, equal ] };
+
+   key <AD01> { [ 0x1000636, 0x1000652, degree ] }; // Arabic dad, Arabic sukun
+   key <AD02> { [ 0x1000635, 0x100064C, 0x1000653 ] }; // Arabic sad, Arabic dammatan, Arabic maddah above
+   key <AD03> { [ 0x100062B, 0x100064D, eurosign ] }; // Arabic theh, Arabic kasratan
+   key <AD04> { [ 0x1000642, 0x100064B, 0x100FD3E ] }; // Arabic qaf, Arabic fathatan, ornate left paren
+   key <AD05> { [ 0x1000641, 0x100064F, 0x100FD3F ] }; // Arabic feh, Arabic damma, ornate right paren
+   key <AD06> { [ 0x100063A, 0x1000650, 0x1000656 ] }; // Arabic ghain, Arabic kasra, Arabic subscript alef
+   key <AD07> { [ 0x1000639, 0x100064E, 0x100e659 ] }; // Arabic ain, Arabic fatha, Arabic zwarakay
+   key <AD08> { [ 0x1000647, 0x1000651, 0x1000670 ] }; // Arabic heh, Arabic shadda, Arabic superscript alef
+   key <AD09> { [ 0x100062E, bracketright, apostrophe ] }; // Arabic khah
+   key <AD10> { [ 0x100062D, bracketleft, quotedbl ] }; // Arabic hah
+   key <AD11> { [ 0x100062C, braceright, 0x1000681 ] }; // Arabic jeem, Arabic hah with hamza above
+   key <AD12> { [ 0x1000686, braceleft, 0x1000685 ] }; // Arabic tcheh, Arabic hah with three dots above
+
+   key <AC01> { [ 0x1000634, 0x1000624, 0x100069A ] }; // Arabic sheen, Arabic waw with hamza above, Arabic seen with dot below and dot above
+   key <AC02> { [ 0x1000633, 0x1000626, 0x10006CD ] }; // Arabic seen, Arabic yeh with hamza above, Arabic yeh with tail
+   key <AC03> { [ 0x10006CC, 0x100064A, 0x1000649 ] }; // Farsi yeh, Arabic yeh, Arabic alef maksura
+   key <AC04> { [ 0x1000628, 0x10006D0, 0x1000643  ] }; // Arabic beh, Arabic e, Arabic kaf
+   key <AC05> { [ 0x1000644, 0x1000623, 0x10006B7 ] }; // Arabic lam, Arabic hamza on alef, Arabic alef with hamza above
+   key <AC06> { [ 0x1000627, 0x1000622, 0x1000671 ] }; // Arabic alef, Arabic madda on alef, Arabic alef wasla
+   key <AC07> { [ 0x100062A, 0x1000629, 0x100067C ] }; // Arabic teh, Arabic teh marbuta, Arabic tteh
+   key <AC08> { [ 0x1000646, guillemontright, 0x10006BC ] }; // Arabic noon, Arabic noon with ring
+   key <AC09> { [ 0x1000645, guillemontleft, 0x10006BA ] }; // Arabic meem, Arabic noon ghunna
+   key <AC10> { [ 0x10006A9, colon, semicolon ] }; // Arabic keheh, 
+   key <AC11> { [ 0x10006AF, 0x100061B, 0x10006AB ] }; // Arabic gaf, Arabic semicolon, Arabic kaf with ring
+
+   key <BKSL> { [ backslash, bar, question ] };
+   key <SPCE> { [ space, 0x100200C, 0x10000A0 ] }; // zero width non-joiner, no break space
+
+   key <AB01> { [ 0x1000638, 0x10006C9, 0x10006D2 ] }; // Arabic zah, Arabic kirghiz yu, Arabic yeh barree
+   key <AB02> { [ 0x1000637, 0x10006C7, 0x1000691 ] }; // Arabic tah, Arabic u, Arabic rreh
+   key <AB03> { [ 0x1000632, 0x1000698, 0x1000696 ] }; // Arabic zain, Arabic jeh,  Arabic reh with dot below and dot above
+   key <AB04> { [ 0x1000631, 0x1000625, 0x1000693 ] }; // Arabic_ra (reh?), Arabic alef with hamza below, Arabic reh with ring
+   key <AB05> { [ 0x1000630, 0x100200C, 0x1000688 ] }; // Arabic_thal, zero width non-joiner, Arabic ddal
+   key <AB06> { [ 0x100062F, 0x1000654, 0x1000689 ] }; // Arabic dal, Arabic hamza above, Arabic dal with ring
+   key <AB07> { [ 0x100067E, 0x1000621, 0x1000679 ] }; // Arabic peh, Arabic hamza, Arabic tteh
+   key <AB08> { [ 0x1000648, greater, comma ] }; // Arabic_waw
+   key <AB09> { [ period, less ] };
+   key <AB10> { [ slash, 0x100061F, 0x1000655 ] }; // Arabic question mark, Arabic hamza below
+
+   //language key
+   key <I219> { [ ISO_Next_Group,  ISO_Prev_Group ] };
+};
Index: rules/base.xml.in
===================================================================
RCS file: /cvs/xlibs/xkbdesc/xkeyboard-config/rules/base.xml.in,v
retrieving revision 1.224
diff -u -p -r1.224 base.xml.in
--- rules/base.xml.in	18 Nov 2007 11:42:54 -0000	1.224
+++ rules/base.xml.in	20 Nov 2007 04:49:22 -0000
@@ -1082,6 +1082,24 @@
             <_description>Southern Uzbek</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>olpc-ps</name>
+            <_description>OLPC Pashto</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>olpc-da</name>
+            <_description>OLPC Dari</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>olpc-uz</name>
+            <_description>OLPC Southern Uzbek</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
