diff -up xkeyboard-config-1.4/symbols/inet.orig xkeyboard-config-1.4/symbols/inet
--- xkeyboard-config-1.4/symbols/inet.orig	2008-10-24 11:49:36.000000000 +1030
+++ xkeyboard-config-1.4/symbols/inet	2008-10-24 11:49:52.000000000 +1030
@@ -158,9 +158,10 @@ xkb_symbols "evdev" {
     key <I241>   {      [ XF86MailForward       ]       };
     key <I242>   {      [ XF86Save              ]       };
     key <I243>   {      [ XF86Documents         ]       };
-//  key <I244>   {      [ ]       }; // KEY_BATTERY
-//  key <I245>   {      [ ]       }; // KEY_BLUETOOTH
-//  key <I246>   {      [ ]       }; // KEY_WLAN
+    key <I244>   {      [ XF86Battery           ]       }; // KEY_BATTERY
+    key <I245>   {      [ XF86Bluetooth         ]       }; // KEY_BLUETOOTH
+    key <I246>   {      [ XF86WLAN              ]       }; // KEY_WLAN
+    key <I247>   {      [ XF86UWB               ]       }; // KEY_UWB
 //  key <I249>   {      [ ]       }; // KEY_VIDEO_NEXT  --  drive next video source
 //  key <I250>   {      [ ]       }; // KEY_VIDEO_PREV  --  drive previous video source
 //  key <I251>   {      [ ]       }; // KEY_BRIGHTNESS_CYCLE  --  bright up, max++ == min
