From 3b0326435e6f02c0d1c92b1c7f9902d81f1245aa Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@redhat.com>
Date: Thu, 5 Mar 2009 13:01:20 +1000
Subject: [PATCH] Map I213 to XF86Suspend, and I255 to XF86Hibernate.

Signed-off-by: Peter Hutterer <peter.hutterer@redhat.com>
---
 symbols/inet |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/symbols/inet b/symbols/inet
index ba55f43..dafb406 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -165,7 +165,7 @@ xkb_symbols "evdev" {
     key <I209>   {      [ XF86AudioPause        ]       };
     key <I210>   {      [ XF86Launch3           ]       };
     key <I211>   {      [ XF86Launch4           ]       };
-    key <I213>   {      [ XF86Standby           ]       };
+    key <I213>   {      [ XF86Suspend           ]       };
     key <I214>   {      [ XF86Close             ]       };
     key <I215>   {      [ XF86AudioPlay         ]       };
     key <I216>   {      [ XF86Forward           ]       };
@@ -205,6 +205,8 @@ xkb_symbols "evdev" {
 //  key <I251>   {      [ ]       }; // KEY_BRIGHTNESS_CYCLE  --  bright up, max++ == min
 //  key <I252>   {      [ ]       }; // KEY_BRIGHTNESS_ZERO  --  brightness off
 //  key <I253>   {      [ ]       }; // KEY_DISPLAY_OFF  --  turn off display
+//  key <I254>   {      [ ]       }; // KEY_WIMAX
+    key <I255>   {      [ XF86Hibernate         ]       }; // KEY_HIBERNATE
  
 };
 
-- 
1.6.0.6

