--- xkeyboard-config-0.8/rules/base.kinesis	2006-08-24 21:39:56.000000000 -0400
+++ xkeyboard-config-0.8/rules/base	2006-08-24 21:40:45.000000000 -0400
@@ -115,6 +115,7 @@
   pc98		=	nec(pc98)
   abnt2		=	pc(abnt2)
  $mac		=	macintosh(macintosh)
+  kinesis       =       kinesis(model100)
   *		=       pc(pc104)
 
 ! model		layout		variant		=	symbols
--- xkeyboard-config-0.8/rules/base.xml.in.kinesis	2006-08-24 21:40:52.000000000 -0400
+++ xkeyboard-config-0.8/rules/base.xml.in	2006-08-24 21:42:13.000000000 -0400
@@ -704,6 +704,12 @@
         <_description>Laptop/notebook eMachines m68xx</_description>
       </configItem>
     </model>
+    <model>
+      <configItem>
+        <name>kinesis</name>
+        <description>Kinesis Classic Keyboard</description>
+      </configItem>
+    </model>
   </modelList>
   <layoutList>
     <layout>
